File: C++/ej1.cpp
#include <iostream>
#include <string>
#include <algorithm>

int main() {
  std::cin.tie(nullptr);
  std::ios_base::sync_with_stdio(false);

  // TODO: fixme.
  //  Leer la cadena de entrada
  std::string cadena;
  std::getline(std::cin, cadena);

  // Invertir la cadena
  std::reverse(cadena.begin(), cadena.end());

  //  Imprimir la cadena invertida
  std::cout << cadena << std::endl;

  return 0;
}

File: JAVA/Ej1.java
import java.util.Scanner;
/**
 *  Entrada: Cadena de prueba       Salida: abeurp ed anedaC
 * @author HP - MSS | InvertirCadena
 */
public class Ej1 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Leer la línea completa
        String cadena = scanner.nextLine();

        // Convertir la cadena a un array de caracteres
        char[] caracteres = cadena.toCharArray();

        // Invertir el array de caracteres
        for (int i = 0, j = caracteres.length - 1; i < j; i++, j--) {
            char temp = caracteres[i];
            caracteres[i] = caracteres[j];
            caracteres[j] = temp;
        }

        // Imprimir la cadena invertida
        System.out.println(new String(caracteres));

        scanner.close();
    }
    
}